app/controllers/api/v2/models_controller.rb

```diff
@@ -47,6 +47,12 @@ def update
       def destroy
         process_response @model.destroy
       end
+
+      private
+
+      def find_resource
+        @model = Model.friendly.find(params[:id]) || super
+      end
     end
```

Comment:
that's about right. names can be numbers for Model. in which case, find_resource will look for ID instead of starting with name,